Sources/TuistSupport/Extensions/AbsolutePath+Extras.swift

```diff
@@ -101,6 +101,17 @@ extension AbsolutePath {
     public func sha256() -> Data? {
         try? SHA256Digest.file(at: url)
     }
+
+    /// Returns target and configuration name from a file name
+    ///
+    /// Expects the file to be named "TargetName.ConfigurationName.extension"
+    ///
+    /// - Returns: Tuple consisting of targetName and configurationName
+    public func extractTargetAndConfigurationName() -> (targetName: String, configurationName: String)? {
```

Comment:
Nitpick: It's a bit odd that a method that is specific to a use-case (signing) live in `AbsolutePath` where all generic methods are defined. I'd recommend moving it to a more scoped place.

Reply:
I know - wasn't happy about that either - what would you recommend - add a `AbsolutePath+SiginingExtras.swift` to the `TuistSigning` folder?

Reply:
I think it'd make sense to delete `extractTargetAndConfigurationName` function. This logic would be duplicated in two places. The other option is to create/add it to some service rather than extension on `AbsolutePath`